### PR TITLE
Polish and rename `.git-blame-ignore-revs` grammar

### PR DESCRIPTION
**Pull request by Alhadis**
This pull-request combines the following changes in the support of github/linguist#5903:

1.	**Grammar renamed to be more context-agnostic.**[^1]
2.	**Revision IDs with leading or trailing whitespace are now highlighted correctly.**
3.	**Invalid lines containing whitespace now yield nicer-looking highlighting.**

[^1]: I [explained](https://github.com/github/linguist/pull/5903/files#r878683609) the rationale for this in my original review, but if you feel strongly about keeping the current name for some reason, you can revert ed99b26 specifically.

**Preview:** [Before](https://novalightshow.netlify.app/?grammar-type=url&grammar=https%3A%2F%2Fraw.githubusercontent.com%2FNixinova%2FNovaGrammars%2F1d92f4f73ed125bcd6912ae3ec804e405f4076a2%2Fgrammars%2Fgit-blame-ignore-revs.yaml-tmLanguage&sample-type=text&sample=%23+40-digit+checksum+%28valid%29%0A1d92f4f73ed125bcd6912ae3ec804e405f4076a2%0A%0A%23+41-digit+checksum+%28invalid%29%0A1d92f4f73ed125bcd6912ae3ec804e405f4076a20%0A%0AInvalid+line+with+whitespace%0A%0A%0A%09%23+Indented%2C+40-digit+checksum+%28valid%29%0A%091d92f4f73ed125bcd6912ae3ec804e405f4076a2%0A%0A%09%23+Indented%2C+41-digit+checksum+%28invalid%29%0A%091d92f4f73ed125bcd6912ae3ec804e405f4076a20%0A%0A+++Indented%2C+invalid+line+with+whitespace+++%0A) | [After](https://novalightshow.netlify.app/?grammar-type=text&grammar=name%3A+Git+Revision+List%0AscopeName%3A+source.git-revlist%0Apatterns%3A%0A-+include%3A+%27%23main%27%0A%0Arepository%3A%0A++main%3A%0A++++patterns%3A%0A++++-+include%3A+%27%23comment%27%0A++++-+include%3A+%27%23sha%27%0A++++-+include%3A+%27%23invalid%27%0A%0A++comment%3A%0A++++name%3A+comment.line.git-revlist%0A++++begin%3A+%27%23%27%0A++++end%3A+%27%24%27%0A++++beginCaptures%3A%0A++++++%270%27%3A%0A++++++++name%3A+punctuation.definition.comment.git-revlist%0A%0A++sha%3A%0A++++name%3A+constant.numeric.sha.git-revlist%0A++++match%3A+%27%5B0-9a-fA-F%5D%7B40%7D%28%3F%3D%5Cs*%24%29%27%0A%0A++invalid%3A%0A++++name%3A+invalid.illegal.git-revlist%0A++++match%3A+%27%5CS.*%3F%28%3F%3D%5Cs*%24%29%27%0A&sample-type=text&sample=%23+40-digit+checksum+%28valid%29%0A1d92f4f73ed125bcd6912ae3ec804e405f4076a2%0A%0A%23+41-digit+checksum+%28invalid%29%0A1d92f4f73ed125bcd6912ae3ec804e405f4076a20%0A%0AInvalid+line+with+whitespace%0A%0A%0A%09%23+Indented%2C+40-digit+checksum+%28valid%29%0A%091d92f4f73ed125bcd6912ae3ec804e405f4076a2%0A%0A%09%23+Indented%2C+41-digit+checksum+%28invalid%29%0A%091d92f4f73ed125bcd6912ae3ec804e405f4076a20%0A%0A+++Indented%2C+invalid+line+with+whitespace+++%0A)

### CODE REVIEW

**Review by Nixinova — CHANGES_REQUESTED** [review]
You've got an erroneous CSON file tagging along there.

**Comment by Alhadis** [comment]
Sorry, fixed.

**Comment by Alhadis** [comment]
@Nixinova BTW, that CSON file was created because I needed to manually add a `fileTypes` array in order to preview it in Atom:

```diff
 name: Git Revision List
 scopeName: source.git-revlist
+fileTypes:
+- git-blame-ignore-revs
 patterns:
 - include: '#main'
```

If you make a point of filling out the `fileTypes` array, it might help to automate conversion to other grammar formats. :wink: Note that Atom stupidly requires leading dots to be removed from both filenames _and_ file-extensions, so for maximum portability, the above snippet should really be:

~~~yaml
fileTypes:
-  git-blame-ignore-revs
- .git-blame-ignore-revs
~~~

No pressure if you CBF though, it's purely on you. Seriously.

File: grammars/git-revlist.yaml-tmLanguage [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Git Revision List
+scopeName: source.git-revlist
+patterns:
```
Comment (Nixinova) [inline grammars/git-revlist.yaml-tmLanguage:3]:
```suggestion
fileTypes:
-  git-blame-ignore-revs
- .git-blame-ignore-revs
patterns:
```

**Review by Nixinova — APPROVED** [review]
Looks good, thank you!

**Pull request merged**